src/Microsoft.Extensions.Caching.Abstractions/CacheEntryExtensions.cs

```diff
@@ -148,7 +136,6 @@ public static ICacheEntry SetOptions(this ICacheEntry entry, MemoryCacheEntryOpt
             entry.AbsoluteExpiration = options.AbsoluteExpiration;
             entry.AbsoluteExpirationRelativeToNow = options.AbsoluteExpirationRelativeToNow;
             entry.SlidingExpiration = options.SlidingExpiration;
-            entry.Priority = options.Priority;
```

Comment:
All components share an eviction strategy anyways, and if the metadata were embedded in the value you'd have the same problem of everyone having to know about it.